stdlib/public/core/StringBridge.swift

```diff
@@ -125,57 +125,34 @@ private var kCFStringEncodingUTF8 : _swift_shims_CFStringEncoding {
   @inline(__always) get { return 0x8000100 }
 }
 
-@_effects(readonly)
-private func _unsafeAddressOfCocoaStringClass(_ str: _CocoaString) -> UInt {
-  return _swift_stdlib_unsafeAddressOfClass(str)
-}
-
-internal enum _KnownCocoaString {
-  case storage
-  case shared
-  case cocoa
-#if !(arch(i386) || arch(arm))
-  case tagged
-#endif
-  
-  @inline(__always)
-  init(_ str: _CocoaString) {
-    
-#if !(arch(i386) || arch(arm))
-    if _isObjCTaggedPointer(str) {
-      self = .tagged
-      return
-    }
-#endif
-    
-    switch _unsafeAddressOfCocoaStringClass(str) {
-    case unsafeBitCast(__StringStorage.self, to: UInt.self):
-      self = .storage
-    case unsafeBitCast(__SharedStringStorage.self, to: UInt.self):
-      self = .shared
-    default:
-      self = .cocoa
-    }
-  }
-}
-
 #if !(arch(i386) || arch(arm))
 // Resiliently write a tagged _CocoaString's contents into a buffer.
 @_effects(releasenone) // @opaque
 internal func _bridgeTagged(
   _ cocoa: _CocoaString,
+  length: Int,
   intoUTF8 bufPtr: UnsafeMutableBufferPointer<UInt8>
 ) -> Int? {
-  _internalInvariant(_isObjCTaggedPointer(cocoa))
   let ptr = bufPtr.baseAddress._unsafelyUnwrappedUnchecked
-  let length = _stdlib_binary_CFStringGetLength(cocoa)
   _internalInvariant(length <= _SmallString.capacity)
   var count = 0
   let numCharWritten = _swift_stdlib_CFStringGetBytes(
     cocoa, _swift_shims_CFRange(location: 0, length: length),
     kCFStringEncodingUTF8, 0, 0, ptr, bufPtr.count, &count)
   return length == numCharWritten ? count : nil
 }
+
+//In the future we may want to try CFStringGetCStringPtr and so on here,
+//or _bridgeTagged may want to use faster tagged-specific accessors than
+//CFStringGetCString, but for now they're the same
+@_effects(releasenone)
+internal func _bridgeNonTagged(
+  _ cocoa: _CocoaString,
+  length: Int,
+  intoUTF8 bufPtr: UnsafeMutableBufferPointer<UInt8>
+  ) -> Int? {
+  return _bridgeTagged(cocoa, length: length, intoUTF8: bufPtr)
```

Comment:
Shouldn't we write this the other way around then? I.e. bridgeTagged calls bridgeNonTagged, but might have fast-paths in the future? That way we can also keep the invariant check.

Reply:
Yeah that sounds reasonable. Habits from not having an inliner; I was trying to keep the fast path from having an extra call in it.